packages/apps/transfer/src/services/transfer/coin-transfer.ts

```diff
@@ -0,0 +1,225 @@
+import { ChainwebNetworkId } from '@kadena/chainweb-node-client';
+import { PactCommand } from '@kadena/client';
+import { sign } from '@kadena/cryptography-utils';
+import { PactNumber } from '@kadena/pactjs';
+import { ChainId } from '@kadena/types';
+
+import { getPublicKeys } from '../../services/accounts/get-public-keys';
+import { generateApiHost } from '../utils/utils';
+
+const gasLimit: number = 2300;
+const gasPrice: number = 0.00001;
+const ttl: number = 28800;
+
+export interface TransferResult {
+  requestKey?: string;
+  status?: string;
+}
+
+export async function coinTransfer({
+  fromAccount,
+  fromChainId,
+  toAccount,
+  toChainId,
+  amount,
+  fromPrivateKey,
+  networkId,
+}: {
+  fromAccount: string;
+  fromChainId: ChainId;
+  toAccount: string;
+  toChainId: ChainId;
+  amount: string;
+  fromPrivateKey: string;
+  networkId: ChainwebNetworkId;
+}): Promise<PactCommand> {
+  if (fromChainId === toChainId) {
+    return transferCreate({
+      fromAccount,
+      toAccount,
+      amount,
+      fromPrivateKey,
+      fromChainId,
+      networkId,
+    });
+  }
+
+  return crossTransfer({
+    fromAccount,
+    fromChainId,
+    toAccount,
+    toChainId,
+    amount,
+    fromPrivateKey,
+    networkId,
+  });
+}
+
+export async function transferCreate({
+  fromAccount,
+  fromChainId,
+  toAccount,
+  amount,
+  fromPrivateKey,
+  networkId,
+}: {
+  fromAccount: string;
+  fromChainId: ChainId;
+  toAccount: string;
+  amount: string;
+  fromPrivateKey: string;
+  networkId: ChainwebNetworkId;
+}): Promise<PactCommand> {
+  if (isNaN(Number(amount))) {
+    throw new Error('Amount must be a number');
+  }
+
+  const fromAccountGuard = await getPublicKeys(
+    networkId,
+    fromChainId,
+    fromAccount,
+  );
+  if (fromAccountGuard === undefined) {
+    throw new Error('Sender account does not exist');
+  }
+
+  const toAccountGuard = await getPublicKeys(networkId, fromChainId, toAccount);
+  if (toAccountGuard === undefined) {
+    throw new Error('Receiver account does not exist');
+  }
+
+  const pactCommand = new PactCommand();
+  pactCommand.code = `(coin.transfer-create "${fromAccount}" "${toAccount}" (read-keyset "ks") ${new PactNumber(
+    amount,
+  ).toDecimal()})`;
```

Comment:
Not sure if we should at this moment, but it could be nice to use `Pact.modules.coin.transfer` here with the generated types?

(Applies to other functions as well)

Reply:
Most of this code will be removed this week since we are not doing coin transfers anymore. If we do still have some of this code by then we will check it out 👍